src/umap.ts

```diff
@@ -809,16 +809,15 @@ export class UMAP {
     const weights: number[] = [];
     const head: number[] = [];
     const tail: number[] = [];
-    for (let i = 0; i < graph.nRows; i++) {
-      for (let j = 0; j < graph.nCols; j++) {
-        const value = graph.get(i, j);
-        if (value) {
-          weights.push(value);
-          tail.push(i);
-          head.push(j);
+    var rowColValues = graph.getAll();
+    for (let i = 0; i < rowColValues.length; i++) {
```

Comment:
Please use 2 spaces per indentation, better yet - use `prettier` to format everything automatically!